channelmonitor/channelmonitor.go

```diff
@@ -440,7 +489,7 @@ func (mc *monitoredChannel) closeChannelAndShutdown(cherr error) {
 // Wait for the peer to send an acknowledgement to the restart request
 func (mc *monitoredChannel) waitForRestartResponse() chan error {
 	restartFired := make(chan struct{})
-	restarted := make(chan error)
+	restarted := make(chan error, 3)
```

Comment:
Buffered chan with size 3? Why is that? I guess it makes sense that we don't want to block when we push to `restarted`, but why 3? Isn't it possible to block when we push to the channel 3 times?

Reply:
We should only get a single restart event, but potentially we could also get a context timeout and a timer expiry.
But actually as soon as we get any of those events we exit the select to it doesn't make sense for it to be 3, I will fix